addons/base_import/models.py

```diff
@@ -126,6 +134,56 @@ def get_fields(self, cr, uid, model, context=None,
         # TODO: cache on model?
         return fields
 
+    def _read_import_file(self, record, options):
+        rows = []
+        if record.file_type == 'text/csv':
+            rows = self._read_csv(record, options)
+        elif record.file_type == 'application/vnd.ms-excel' or record.file_type == 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet':
+            if xlrd is None:
+                raise UserError(_(" XLRD library not found. XLS/XLSX file would not import."))
+            else:
+                rows = self._read_xls_xlsx(record, options)
+        elif record.file_type == 'application/vnd.oasis.opendocument.spreadsheet':
+            rows = self._read_ods(record, options)
+        return rows
```

Comment:
Why is there a UserError raised if trying to import an XLS/XLSX file but no information when trying to import an unkwown data type, just an empty list of rows?